.github/workflows/incremental-deploy.yml

```
@@ -22,13 +29,17 @@ jobs:
           fetch-depth: 0
       - name: Authenticate target org
         run: |
-          sfdx org login sfdx-url --set-default --sfdx-url-file <(echo "${{ secrets.SFDX_AUTH_URL_TARGET_ORG }}")
+          sfdx org login sfdx-url --set-default --sfdx-url-file <(echo "${{ inputs.sfdxAuthUrl }}")
       - name: Deploy changes to target org
-        # Note: This requires the merged PR to only have a single commit or merge commit
         run: |
-          sfdx sgd:source:delta --from "HEAD~1" --to "HEAD" --output "." --ignore .forceignore
+          git merge "${{ inputs.baseRef }}"
```

I would say that this should only be done if `validateOnly` is true, otherwise (i.e in a real deployment) there is no need to refresh the current branch. Don't you think? I think it could be safer/less confusing.

```suggestion
    if [ "{{  inputs. validateOnly  }}" = "true" ]; then
          git merge "${{ inputs.baseRef }}"
    fi
```